components/nexus-commands/src/main/java/org/sonatype/nexus/commands/CommandSupport.java

```diff
@@ -17,30 +17,55 @@
 
 import javax.annotation.Nullable;
 
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.karaf.shell.commands.CommandWithAction;
 import org.apache.karaf.shell.commands.basic.AbstractCommand;
 import org.apache.karaf.shell.console.CompletableFunction;
 import org.apache.karaf.shell.console.Completer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * Helper class that ties {@link AbstractCommand} and {@link CompletableFunction} together.
+ * Provides support for Karaf {@link CommandWithAction} implementations.
  * 
  * @since 3.0
  */
-public abstract class AbstractCompletableCommand
+public abstract class CommandSupport
     extends AbstractCommand
     implements CompletableFunction
 {
+  protected final Logger log = LoggerFactory.getLogger(getClass());
+
+  private List<Completer> argumentCompleters;
+
+  private Map<String, Completer> optionCompleters;
+
+  protected void addArgumentCompleter(final int index, final Completer completer) {
+    if (argumentCompleters == null) {
```

Comment:
Mostly because there seems to be logic in the upper handling of completer muck that does something specifically for null, translates to NullCompleter.INSTANCE.  That and avoiding making empty collections where this feature is not used.